### ISSUE TITLE
Amazon S3 TransferManager not running.

### ISSUE THREAD

**Issue by tonyjhang** [WowzaMediaSystems/wse-plugin-s3upload#15]
When I use this plugin , it show the error message below:
-----------------------------------------------------------------------------------------------------------------
WARN	server	comment	2018-06-13	09:05:07	-	-	-	-	-	87.101	-	-	-	-	-	-	-	-	ModuleS3Upload.WriteListener.onWriteComplete Cannot upload file because S3 Transfer Manager isn't loaded: [live/_definst_/openlab.stream.mp4]

WARN	server	comment	2018-06-13	09:05:07	-	-	-	-	-	87.107	-	-	-	-	-	-	-	-	ModuleS3Upload.startUpload problem starting or resuming upload: [live/_definst_/openlab.stream.mp4.upload] Amazon S3 TransferManager not running.
------------------------------------------------------------------------------------------------------------------
My Wowza Stream Engine PAID is on AWS version is 4.7.5.
OS:Windows 
AWS SDK: aws-java-sdk-1.11.346 and 1.10.77 have tried
Media file prefix:C:\Program Files (x86)\Wowza Media Systems\Wowza Streaming Engine\content

**Comment by rogerlittin** [comment]
You would receive that message if there was a problem with the aws authentication. You may have other log messages that indicate what the problem is.

**Comment by tonyjhang** [comment]
It's my all error message.My AWS SDK version is 1.10.77 in this case . 
The configuration and parameter I config is all reference below. I tried it two month ago ,and it's can worked at that time . 
https://www.wowza.com/docs/how-to-upload-recorded-media-to-an-amazon-s3-bucket-modules3upload
-------------------------------------------------------------------------------------------------------------------
#Version: 1.0
#Start-Date: 2018-06-14 02:24:38 UTC
#Software: Wowza Streaming Engine 4.7.5 build21763
#Date: 2018-06-14
#Fields: x-severity	x-category	x-event	date	time	c-client-id	c-ip	c-port	cs-bytes	sc-bytes	x-duration	x-sname	x-stream-id	x-spos	sc-stream-bytes	cs-stream-bytes	x-file-size	x-file-length	x-ctx	x-comment
ERROR	server	comment	2018-06-14	02:25:56	-	-	-	-	-	78.493	-	-	-	-	-	-	-	-	ModuleS3Upload.onAppStart [live/_definst_] The installed version of AWS SDK isn't compatible with this version of Wowza Streaming Engine. Please upgrade your version of AWS SDK
WARN	server	comment	2018-06-14	02:25:58	-	-	-	-	-	80.367	-	-	-	-	-	-	-	-	CupertinoPacketHandler.handleHolder[live/_definst_/openlab.stream]: Invalid video/audio codec combination for iPhone/iPod: video:H264 audio:G711_MULAW
WARN	server	comment	2018-06-14	02:25:58	-	-	-	-	-	80.368	-	-	-	-	-	-	-	-	CupertinoPacketHandler.handleHolder[live/_definst_/openlab.stream]: Invalid audio codec for iPhone/iPod:G711_MULAW
WARN	server	comment	2018-06-14	02:26:10	-	-	-	-	-	92.692	-	-	-	-	-	-	-	-	ModuleS3Upload.WriteListener.onWriteComplete Cannot upload file because S3 Transfer Manager isn't loaded: [live/_definst_/openlab.stream.mp4]
WARN	server	comment	2018-06-14	02:26:10	-	-	-	-	-	92.696	-	-	-	-	-	-	-	-	ModuleS3Upload.startUpload problem starting or resuming upload: [live/_definst_/openlab.stream.mp4.upload] Amazon S3 TransferManager not running.
WARN	server	comment	2018-06-14	02:26:11	-	-	-	-	-	93.883	-	-	-	-	-	-	-	-	CupertinoPacketHandler.handleHolder[live/_definst_/openlab.stream]: Invalid video/audio codec combination for iPhone/iPod: video:H264 audio:G711_MULAW
WARN	server	comment	2018-06-14	02:26:11	-	-	-	-	-	93.899	-	-	-	-	-	-	-	-	CupertinoPacketHandler.handleHolder[live/_definst_/openlab.stream]: Invalid audio codec for iPhone/iPod:G711_MULAW
WARN	server	comment	2018-06-14	02:26:12	-	-	-	-	-	94.323	-	-	-	-	-	-	-	-	LiveStreamPacketizerPacketHandler.handlePacket[live/_definst_/openlab.stream]: Audio and video codecs cannot be packetized together in a single stream: audio:G711_MULAW video:H264
WARN	server	comment	2018-06-14	02:26:12	-	-	-	-	-	94.734	-	-	-	-	-	-	-	-	[live/_definst_/openlab.stream] LiveStreamRecorderMP4.work() Unsupported audio format for MP4 container: CODEC_AUDIO_G711_MULAW
WARN	server	comment	2018-06-14	02:26:26	-	-	-	-	-	108.795	-	-	-	-	-	-	-	-	ModuleS3Upload.WriteListener.onWriteComplete Cannot upload file because S3 Transfer Manager isn't loaded: [live/_definst_/openlab.stream_0.mp4]
WARN	server	comment	2018-06-14	02:26:26	-	-	-	-	-	108.807	-	-	-	-	-	-	-	-	ModuleS3Upload.startUpload problem starting or resuming upload: [live/_definst_/openlab.stream_0.mp4.upload] Amazon S3 TransferManager not running.

**Comment by rogerlittin** [comment]
Hi,

As the first error message states, `The installed version of AWS SDK isn't compatible with this version of Wowza Streaming Engine.` Your version of Wowza Streaming Engine requires AWS SDK version 1.11.xxx or later. Please remove the existing version and install the latest version and test again.

Regards,

Roger.

**Comment by tonyjhang** [comment]
Hi, I also try AWS SDK 1.11.346 by using aws-java-sdk-1.11.346.jar ,but it's shows the same message.
In this message:
ModuleS3Upload.WriteListener.onWriteComplete Cannot upload file because S3 Transfer Manager isn't loaded: [live/definst/openlab.stream_0.mp4]
Is it possible when wowza try to upload file , but it input error path cause exception? 

Best Regards,
Tony Jhang

**Comment by rogerlittin** [comment]
Hi Tony,

Please post the full logs like you did before. The previous logs indicated that the problem was related to the old version. This time there will probably be a different problem.

**Comment by tonyjhang** [comment]
In AWS SDK 1.11.346 , it's show the same message in my VM(OS:Windows).
My media file folder create .upload file but it's doesn't work.
The message below is my error message using the latest AWS SDK
----------------------------------------------------------------------------------------------------
#Version: 1.0
#Start-Date: 2018-06-14 05:25:58 UTC
#Software: Wowza Streaming Engine 4.7.5 build21763
#Date: 2018-06-14
#Fields: x-severity	x-category	x-event	date	time	c-client-id	c-ip	c-port	cs-bytes	sc-bytes	x-duration	x-sname	x-stream-id	x-spos	sc-stream-bytes	cs-stream-bytes	x-file-size	x-file-length	x-ctx	x-comment
ERROR	server	comment	2018-06-14	05:27:01	-	-	-	-	-	63.513	-	-	-	-	-	-	-	-	ModuleS3Upload.onAppStart [live/_definst_] The installed version of AWS SDK isn't compatible with this version of Wowza Streaming Engine. Please upgrade your version of AWS SDK
WARN	server	comment	2018-06-14	05:27:03	-	-	-	-	-	65.338	-	-	-	-	-	-	-	-	CupertinoPacketHandler.handleHolder[live/_definst_/openlab.stream]: Invalid video/audio codec combination for iPhone/iPod: video:H264 audio:G711_MULAW
WARN	server	comment	2018-06-14	05:27:03	-	-	-	-	-	65.347	-	-	-	-	-	-	-	-	CupertinoPacketHandler.handleHolder[live/_definst_/openlab.stream]: Invalid audio codec for iPhone/iPod:G711_MULAW
WARN	server	comment	2018-06-14	05:27:04	-	-	-	-	-	65.882	-	-	-	-	-	-	-	-	LiveStreamPacketizerPacketHandler.handlePacket[live/_definst_/openlab.stream]: Audio and video codecs cannot be packetized together in a single stream: audio:G711_MULAW video:H264
WARN	server	comment	2018-06-14	05:27:08	-	-	-	-	-	70.139	-	-	-	-	-	-	-	-	[live/_definst_/openlab.stream] LiveStreamRecorderMP4.work() Unsupported audio format for MP4 container: CODEC_AUDIO_G711_MULAW
WARN	server	comment	2018-06-14	05:27:45	-	-	-	-	-	107.853	-	-	-	-	-	-	-	-	ModuleS3Upload.WriteListener.onWriteComplete Cannot upload file because S3 Transfer Manager isn't loaded: [live/_definst_/openlab.stream.mp4]
WARN	server	comment	2018-06-14	05:27:45	-	-	-	-	-	107.854	-	-	-	-	-	-	-	-	ModuleS3Upload.startUpload problem starting or resuming upload: [live/_definst_/openlab.stream.mp4.upload] Amazon S3 TransferManager not running.

**Comment by rogerlittin** [comment]
Hi,

I've just confirmed that version should be working fine. The message is still indicating that you have an incompatible version. Please make sure you have removed the old versions of the AWS SDK from your lib folder and that you have restarted Wowza Streaming Engine.

-Roger.

**Comment by tonyjhang** [comment]
Hi,
In my lib folder , I add these below:
aws-java-sdk-1.11.346.jar
wse-plugin-s3upload.jar
In lib folder , it only add this SDK file.
After that I restart Wowza Stream Engine , it's also show the same message . 
Is it something I need to configure?

Tony

**Comment by rogerlittin** [comment]
Can you provide a listing of the lib folder?

**Comment by tonyjhang** [comment]
These jar file download from:
https://www.wowza.com/docs/how-to-upload-recorded-media-to-an-amazon-s3-bucket-modules3upload

My lib folder below: 

Volume in drive C has no label.
 Volume Serial Number is 3A93-13CD

 Directory of C:\Program Files (x86)\Wowza Media Systems\Wowza Streaming Engine\lib

06/14/2018  06:42 AM    <DIR>          .
06/14/2018  06:42 AM    <DIR>          ..
**06/12/2018  03:27 PM        89,534,805 aws-java-sdk-1.11.346.jar**
04/02/2018  09:00 PM           646,979 azure-storage-2.2.0.jar
04/02/2018  09:00 PM         4,092,400 bcprov-jdk15on-159.jar
04/02/2018  09:00 PM           284,220 commons-lang-2.6.jar
04/02/2018  09:00 PM           434,678 commons-lang3-3.4.jar
06/14/2018  06:42 AM               159 dir.txt
04/02/2018  09:00 PM           747,794 httpclient-4.5.3.jar
04/02/2018  09:00 PM           323,824 httpcore-4.4.6.jar
04/02/2018  09:00 PM            66,519 jackson-annotations-2.9.0.jar
04/02/2018  09:00 PM           320,134 jackson-core-2.9.0.jar
04/02/2018  09:00 PM         1,328,192 jackson-databind-2.9.0.jar
04/02/2018  09:00 PM            92,142 jackson-dataformat-csv-2.9.0.jar
04/02/2018  09:00 PM            84,007 jackson-dataformat-smile-2.9.0.jar
04/02/2018  09:00 PM            96,721 jackson-dataformat-xml-2.9.0.jar
04/02/2018  09:00 PM            40,369 jackson-dataformat-yaml-2.9.0.jar
04/02/2018  09:00 PM           236,513 jid3lib-0.5.4.jar
04/02/2018  09:00 PM           581,571 joda-time-2.3.jar
04/02/2018  09:00 PM           489,883 log4j-1.2.17.jar
04/02/2018  09:00 PM            25,962 slf4j-api-1.6.4.jar
04/02/2018  09:00 PM             9,748 slf4j-log4j12-1.6.4.jar
04/02/2018  09:00 PM           473,413 snmp4j-2.4.3.jar
04/02/2018  09:00 PM           621,085 snmp4j-agent-2.4.2.jar
04/02/2018  09:00 PM             1,442 wms-ant.jar
04/02/2018  09:00 PM            31,064 wms-bootstrap.jar
04/02/2018  09:00 PM            87,228 wms-commons-modeler.jar
04/02/2018  09:00 PM           357,582 wms-core.jar
04/02/2018  09:00 PM           141,282 wms-drm.jar
04/02/2018  09:00 PM           136,593 wms-dvr.jar
04/02/2018  09:00 PM           442,769 wms-httpstreamer-cupertinostreaming.jar
04/02/2018  09:00 PM            11,056 wms-httpstreamer-directstreaming.jar
04/02/2018  09:00 PM           394,994 wms-httpstreamer-mpegdashstreaming.jar
04/02/2018  09:00 PM           223,068 wms-httpstreamer-sanjosestreaming.jar
04/02/2018  09:00 PM           277,943 wms-httpstreamer-smoothstreaming.jar
04/02/2018  09:00 PM            34,518 wms-httpstreamer-tsstreaming.jar
04/02/2018  09:00 PM            34,787 wms-httpstreamer-webmstreaming.jar
04/02/2018  09:00 PM           274,913 wms-jax.validation.jar
04/02/2018  09:00 PM            19,617 wms-jlayer.jar
04/02/2018  09:00 PM            18,628 wms-license-rest.jar
04/02/2018  09:00 PM           260,071 wms-mediacache.jar
04/02/2018  09:00 PM            35,279 wms-mediareader-flv.jar
04/02/2018  09:00 PM           161,590 wms-mediareader-h264.jar
04/02/2018  09:00 PM            23,304 wms-mediareader-mp3.jar
04/02/2018  09:00 PM            15,552 wms-mediareader-smil.jar
04/02/2018  09:00 PM            25,725 wms-mediareader-webm.jar
04/02/2018  09:00 PM             8,851 wms-mediawriter-flv.jar
04/02/2018  09:00 PM            95,598 wms-mediawriter-h264.jar
04/02/2018  09:00 PM             8,694 wms-mediawriter-webm.jar
04/05/2018  07:38 PM             7,270 wms-plugin-AmazonAWS.jar
02/21/2012  07:17 PM            33,891 wms-plugin-loadbalancer.jar
12/17/2011  07:30 AM            10,981 wms-plugin-streamnamealias.jar
04/02/2018  09:00 PM           819,923 wms-pushpublish.jar
04/02/2018  09:00 PM            21,978 wms-release-tools.jar
04/02/2018  09:00 PM           930,882 wms-rest.jar
04/02/2018  09:00 PM         2,217,493 wms-restlet-2.2.2.jar
04/02/2018  09:00 PM            11,381 wms-restlet-2.2.2.org.restlet.ext.simple.jar
04/02/2018  09:00 PM         8,087,715 wms-server.jar
04/02/2018  09:00 PM            44,025 wms-sourcecontrol.jar
04/02/2018  09:00 PM             9,477 wms-srt.jar
04/02/2018  09:00 PM            69,131 wms-stats.jar
04/02/2018  09:00 PM            52,692 wms-stream-file.jar
04/02/2018  09:00 PM           127,834 wms-stream-live.jar
04/02/2018  09:00 PM            47,490 wms-stream-publish.jar
04/02/2018  09:00 PM            32,645 wms-stream-record.jar
04/02/2018  09:00 PM            16,186 wms-stream-rtp.jar
04/02/2018  09:00 PM            13,042 wms-stream-shoutcast.jar
04/02/2018  09:00 PM           372,723 wms-transcoder.jar
04/02/2018  09:00 PM           149,680 wms-webrtc.jar
04/02/2018  09:00 PM           538,205 wms-xstream-1.4.7.jar
**06/11/2018  09:46 AM            18,854 wse-plugin-s3upload.jar**
              69 File(s)    117,286,764 bytes
               2 Dir(s)   9,310,314,496 bytes free

**Comment by rogerlittin** [comment]
Hi,

The only difference is that I'm testing with a Linux AMI instead of a Windows AMI. That shouldn't make any difference. Can you please open a support ticket so we can investigate. You can open a ticket from the [Wowza portal](https://store.wowza.com/portal/help/engine). Please provide as much detail as possible, including, zipping up your conf, logs and lib folders.

Thanks,

Roger.

**Comment by tonyjhang** [comment]
OK, I've opened a ticket . 
Thank for your help!
Have a good day :)

**Comment by rogerlittin** [comment]
Hi Tony,

The message is being generated from an exception handler that is triggered when the SDK version is too old. There may be some other reason the exception is being triggered so I've added some more context to the logging. Can you please update to build 51 and retest and post your results to Andrew in the ticket.

Thanks,

Roger.

**Comment by tonyjhang** [comment]
Hi ,
It's worked for your new commit . 
I think this issue can be closed.
Many thanks 👍 

Regards,
Tony

**Comment by rogerlittin** [comment]
The exception handler that we were using to detect the old SDK version was being triggered by a completely unrelated issue. I've fixed that issue and also fixed the exception handler so it will trigger properly.
